client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,379 @@
+	pinterest: 'p:domain_verify',
+	yandex: 'yandex-verification'
+};
+
+function getGeneralSettingsUrl( slug ) {
+	return `/settings/general/${ slug }`;
+}
+
+function stateForSite( site ) {
+	return {
+		seoMetaDescription: get( site, 'options.seo_meta_description', '' ),
+		verificationServicesCodes: {
+			google: get( site, 'options.verification_services_codes.google', '' ),
+			bing: get( site, 'options.verification_services_codes.bing', '' ),
+			pinterest: get( site, 'options.verification_services_codes.pinterest', '' ),
+			yandex: get( site, 'options.verification_services_codes.yandex', '' )
+		}
+	};
+}
+
+function getMetaTagForService( serviceName = '', content = '' ) {
+	if ( ! content ) {
+		return '';
+	}
+
+	if ( content.indexOf( '<meta' ) !== -1 ) {
+		// We were passed a meta tag already!
+		return content;
+	}
+
+	return `<meta name="${ get( serviceIds, serviceName, '' ) }" content="${ content }" />`;
+}
+
+function isValidVerificationCode( serviceName = '', content = '' ) {
+	if ( content.length === 0 ) {
+		return true;
+	}
+
+	content = getMetaTagForService( serviceName, content );
+
+	return content.indexOf( serviceIds[ serviceName ] ) > -1;
+}
+
+export const SeoForm = React.createClass( {
+	displayName: 'SiteSettingsFormSEO',
+
+	mixins: [ protectForm.mixin ],
+
+	getInitialState() {
+		return stateForSite( this.props.site );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( get( nextProps, 'site.ID' ) !== get( this.props, 'site.ID' ) ) {
+			if ( get( nextProps, 'site.jetpack' ) ) {
+				// Go back to general settings if switched to a Jetpack site
+				page( getGeneralSettingsUrl( get( nextProps, 'site.slug', '' ) ) );
+				return;
+			}
+
+			// Update state when switching sites
+			this.setState( stateForSite( nextProps.site ) );
+		}
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seoMetaDescription: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event, serviceName ) {
+		const servicesCodes = this.state.verificationServicesCodes;
+		if ( ! servicesCodes.hasOwnProperty( serviceName ) ) {
+			return;
+		}
+
+		// Show an error if the user types into the field
+		if ( event.target.value.length === 1 ) {
+			this.setState( { showPasteError: true } );
+			return;
+		}
+
+		this.setState( {
+			invalidCodes: [],
+			showPasteError: false,
+			verificationServicesCodes: Object.assign( {}, servicesCodes, {
+				[ serviceName ]: event.target.value
+			} )
+		} );
+	},
+
+	submitSeoForm( event ) {
+		const { site, trackSubmission } = this.props;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		const filteredCodes = pickBy( this.state.verificationServicesCodes, isString );
+
+		const invalidCodes = [];
+		for ( const serviceName in filteredCodes ) {
+			if ( ! isValidVerificationCode( serviceName, filteredCodes[ serviceName ] ) ) {
+				invalidCodes.push( serviceName );
+			}
+		}
```

Comment:
@roundhill if you want to end up with the keys, you can go back to using `pickBy` instead of filter, which returns a new object, then take the `Object.keys()` of the resultant object. it's a bit weird that we can't chain these functions cleanly (we _can_ chain them with lodash), but it's not too bad.

```js
// use previous isInvalid
const invalidCodes = Object.keys( pickBy( verificationServiceCodes, isInvalid ) );
```